RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSCProducer.cc

```diff
@@ -0,0 +1,164 @@
+#include "RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSCProducer.h"
+#include "RecoParticleFlow/PFClusterTools/interface/PFClusterWidthAlgo.h"
+#include "DataFormats/Common/interface/RefToPtr.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/EgammaReco/interface/SuperCluster.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecTrackFwd.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <iostream>
+
+LowPtGsfElectronSCProducer::LowPtGsfElectronSCProducer( const edm::ParameterSet& cfg )
+{
+  produces<SuperClusters>();
+  produces< edm::ValueMap<reco::SuperClusterRef> >();
+  gsfPfRecTracks_ = consumes<GsfPFRecTracks>( cfg.getParameter<edm::InputTag>("gsfPfRecTracks") );
+  ecalClusters_ = consumes<PFClusters>( cfg.getParameter<edm::InputTag>("ecalClusters") );
+  hcalClusters_ = consumes<PFClusters>( cfg.getParameter<edm::InputTag>("hcalClusters") );
+}
+
+LowPtGsfElectronSCProducer::~LowPtGsfElectronSCProducer()
+{}
+
+void LowPtGsfElectronSCProducer::produce( edm::Event& event, const edm::EventSetup& setup )
+{
+
+  edm::Handle<PFClusters> ecalClusters;
+  event.getByToken(ecalClusters_,ecalClusters);
+  if ( !ecalClusters.isValid() ) { edm::LogError("Problem with ecalClusters handle"); }
+
+  edm::Handle<PFClusters> hcalClusters;
+  event.getByToken(hcalClusters_,hcalClusters);
+  if ( !hcalClusters.isValid() ) { edm::LogError("Problem with hcalClusters handle"); }
+
+  edm::Handle<GsfPFRecTracks> gsfPfRecTracks;
+  event.getByToken(gsfPfRecTracks_,gsfPfRecTracks);
+  if ( !gsfPfRecTracks.isValid() ) { edm::LogError("Problem with gsfPfRecTracks handle"); }
+
+  // SuperCluster container and getRefBeforePut
+  std::unique_ptr<SuperClusters> clusters( new SuperClusters() );
```

Comment:
it is better practice to use std::make_unique<SuperClusters>(). The reason for this that its guaranteed not to leak while under very specific circumstances you can leak using it this way. It makes little real  difference but is just better to always use std::make_unique as a rule (except in the few very specific cases you cant).